### PR TITLE
Fixing bugs from PE-D and update some code quality stuff

### PR DESCRIPTION
**Pull request by limyuquan**
The new diagram for DG is the follow:
Screenshot 2024-11-09 at 2 10 16 PM
Updated the association of Car to person
Made Issue connected to Car instead of Person

### CODE REVIEW

**Review by Meowloid — APPROVED** [review]
LGTM

**Review by SeanWong2 — COMMENTED** [review]
LGTM

**Review by calebyyy — APPROVED** [review]
looks good

File: docs/UserGuide.md [hunk 1]
```diff
@@ -536,6 +572,12 @@ The `correct as of` timestamp at the top right corner of the `view` window shows
 </box>
 
 
+<box type="info" seamless>
+
+**Workaround:** If certain details are obstructed in the `view` window, due to unusually large number of `issues`, users can close the window and reopen it using the `view` command to fix the issue. This issue is a planned enhancement for future versions of MATER.
```
Comment (Meowloid) [inline docs/UserGuide.md:577]: bobian bah, thanks for asking prof

**Review by Meowloid — APPROVED** [review]
LGTM.

File: docs/UserGuide.md (hunk 1 above)
Comment (SeanWong2) [reply docs/UserGuide.md:577]: thanks sploof

File: docs/UserGuide.md [hunk 2]
```diff
@@ -240,26 +245,33 @@ add-client n/NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [vrn/CAR_VRN] [vin/CAR_VIN] [
...
 **Notes:**
 
-- If adding a client with a car, all car fields (`vrn`, `vin`, `make`, `model`) must be provided. `vrn` is validated by checksum and invalid `vrn` provided will be rejected by MATER.
+- When adding a client with a car, all car fields (`vrn`, `vin`, `make`, `model`) must be provided.
+- `vrn` is validated using a [checksum letter](#glossary). Invalid `vrn` provided will be rejected by MATER.
 - If the client does not have a car, these fields should be omitted completely.
 
 <box type="tip" seamless>
 
+If you are having trouble entering the `vrn` field or unsure about the correct checksum letter for the `vrn` , you can use the following [link](https://carplatemart.sg/simple-checksum/) to verify if checksum letter for the `vrn` is correct.
```
Comment (xhamyo) [inline docs/UserGuide.md:266]: I think there is an there is an extra space between`vrn`  and `,`

File: docs/UserGuide.md [hunk 3]
```diff
@@ -322,13 +340,19 @@ add-car INDEX vrn/CAR_VRN vin/CAR_VIN make/CAR_MAKE model/CAR_MODEL
...
+| `model/CAR_MODEL` | Required; First character is a capitalised letter or a number; Must be at most 40 characters; | Car's model.                                                       |
 
 **Notes:**
 
 - `vrn` is validated by checksum and invalid `vrn` provided will be rejected by MATER.
 - The client must not already have a car.
 
+<box type="tip" seamless>
+
+If you are having trouble entering the `vrn` field or unsure about the correct checksum letter for the `vrn` , you can use the following [link](https://carplatemart.sg/simple-checksum/) to verify if checksum letter for the `vrn` is correct.
```
Comment (xhamyo) [inline docs/UserGuide.md:352]: I think there is an there is an extra space between `vrn` and `,`

File: docs/UserGuide.md [hunk 4]
```diff
@@ -397,6 +427,12 @@ edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [vrn/CAR_VRN] [vin/CAR_VIN]
 - When editing issues, the existing issues of the client will be replaced with the new issues.
 - To remove all issues from the client, type `i/` without specifying any issues after it.
 
+<box type="tip" seamless>
+
+If you are having trouble entering the `vrn` field or unsure about the correct checksum letter for the `vrn` , you can use the following [link](https://carplatemart.sg/simple-checksum/) to verify if checksum letter for the `vrn` is correct.
```
Comment (xhamyo) [inline docs/UserGuide.md:432]: I think there is an there is an extra space between `vrn` and `,`

**Pull request merged**